network/transport/grpc/tls_offloading.go

```diff
@@ -49,17 +49,23 @@ type tlsOffloadingAuthenticator struct {
 }
 
 func (t *tlsOffloadingAuthenticator) intercept(srv interface{}, serverStream grpc.ServerStream, _ *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	// Det certificate from header
```

Comment:
more like a comment for the type itself

Reply:
The name of this thing is weird to me. It does not authenticate anything, it just extracts the certificate from the header and sets in on the grpc.Peer.